src/phone_number.rs

```diff
@@ -243,45 +243,61 @@ impl<'a> Deref for Country<'a> {
 
 #[cfg(test)]
 mod test {
-    use crate::country;
-    use crate::parser;
-
-    #[test]
-    fn country_id() {
-        assert_eq!(
-            country::AU,
-            parser::parse(None, "+61406823897")
-                .unwrap()
-                .country()
-                .id()
-                .unwrap()
-        );
+    use crate::country::{self, *};
+    use crate::{parser, Mode, PhoneNumber};
+    use anyhow::Context;
+    use rstest::rstest;
+    use rstest_reuse::{self, *};
+
+    fn parsed(number: &str) -> PhoneNumber {
+        parser::parse(None, number)
+            .with_context(|| format!("parsing {number}"))
+            .unwrap()
+    }
 
+    #[template]
+    #[rstest]
+    #[case(parsed("+61406823897"), AU)]
+    #[case(parsed("+32474091150"), BE)]
+    #[case(parsed("+34666777888"), ES)]
+    #[case(parsed("+13459492311"), KY)]
+    #[case(parsed("+16137827274"), CA)]
+    #[case(parsed("+1 520 878 2491"), US)]
+    #[case(parsed("+1-520-878-2491"), US)]
+    // #[case(parsed("+1 520-878-2491"), US)]
```

Comment:
can we maybe leave this one as `#[should_panic]` such that we see the moment we fix it?

Reply:
Didn't immediately see how you can do that with `rstest`, that's why I did it like this.

Reply:
I have added links to the respective issues for it.

Reply:
Let's merge it as-is. The `#[should_panic]` attribute will only add noise to future fixes :-)